include/dlaf/eigensolver/tridiag_solver/tile_collector.h

```diff
@@ -50,13 +50,13 @@ class TileCollector {
   template <class T, Device D>
   auto read(Matrix<const T, D>& mat) const {
     auto [begin, size] = iteratorLocal(mat.distribution());
-    return matrix::util::collectReadTiles(begin, size, mat);
+    return matrix::selectRead(mat, common::iterate_range2d(begin, size));
```

Comment:
To be clear, you're not expecting this to compile now are you? Are you think `IterableRange2D` to be constructible from the local number of tiles?

Reply:
sorry, I meant `common::iterate_range2d(dist.localNrTiles()))`, but it is anyway wrong because I forgot about the fact that `TileCollector` works just on the range and not on the full matrix. 😉